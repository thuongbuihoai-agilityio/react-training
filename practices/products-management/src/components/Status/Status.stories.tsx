import { ComponentStory, ComponentMeta } from '@storybook/react';
import Status, { StatusType } from '.';

export default {
  /* ðŸ‘‡ The title prop is optional.
   * See https://storybook.js.org/docs/react/configure/overview#configure-story-loading
   * to learn how to generate automatic titles
   */
  title: 'Component/Status',
  component: Status
} as ComponentMeta<typeof Status>;

const TemplateStatus: ComponentStory<typeof Status> = args => <Status {...args} />;

export const StatusAvailable = TemplateStatus.bind({});
StatusAvailable.args = {
  value: "Available",
  type: StatusType.primary
};

export const StatusSoldOut = TemplateStatus.bind({});
StatusSoldOut.args = {
  value: "Sold out",
  type: StatusType.secondary
};

export const StatusQuantity = TemplateStatus.bind({});
StatusQuantity.args = {
  value: "9177",
  type: StatusType.tertiary
};
