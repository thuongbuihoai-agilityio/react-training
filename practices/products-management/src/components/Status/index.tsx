import React, { memo } from 'react';
import styles from './Status.module.css';

export enum StatusType {
  default = '',
  primary = 'primary',
  secondary = 'secondary',
  tertiary = 'tertiary'
}

interface StatusLabel {
  value?: boolean;
  quantity?: number;
  type?: string;
  className?: string;
}

const Status: React.FC<StatusLabel> = ({
  type = StatusType.default,
  value = '',
  quantity = 0,
  className = ''
}) => {
  return (
    <div
      data-testid='status'
      className={`${className} ${styles[`status-${type}`]} 
      ${
        quantity
          ? styles[`status-${type}`]
          : value
          ? styles['status-primary']
          : styles['status-secondary']
      }`}
    >
      {quantity ? quantity : `${value ? 'Available' : 'Sold out'}`}
    </div>
  );
};

export default memo(Status);
