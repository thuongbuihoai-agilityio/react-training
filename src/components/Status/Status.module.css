.status-primary,
.status-secondary,
.status-tertiary {
  text-align: center;
  border-radius: 4px;
  padding: 5px 10px;
}

.status-primary {
  color: #7FC008;
  border: 1px solid #7FC008;
  width: 78px;
}

.status-secondary {
  color: #DB303F;
  border: 1px solid #DB303F;
  width: 73px;
}

.status-tertiary {
  background: #fff;
  border: 1px solid #DFE2E9;
  border-radius: 8px;
  width: 57px;
}
