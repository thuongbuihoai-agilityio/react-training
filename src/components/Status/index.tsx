import React, { memo } from 'react';
import styles from './Status.module.css';

export enum StatusType {
  default = '',
  primary = 'primary',
  secondary = 'secondary',
  tertiary = 'tertiary'
}

interface StatusLabel {
  value?: string;
  type?: string;
  className?: string;
}

const Status: React.FC<StatusLabel> = ({
  type = StatusType.default,
  value = '',
  className = '',
}) => {
  return (
    <div data-testid='status' className={`${className} ${styles[`status-${type}`]}`}>
      {value}
    </div>
  );
};

export default memo(Status);
