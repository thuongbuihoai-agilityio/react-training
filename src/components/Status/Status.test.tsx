import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Status, { StatusType } from '.';

describe('Status component', () => {
  test('Should render Status component with size is normal', () => {
    const { getByTestId } = render(
      <Status value='Available' type={StatusType.primary} />
    );
    const status = getByTestId('status');
    expect(status).toBeInTheDocument();
  });

  test('Matches snapshot', () => {
    const { container } = render(<Status />);
    expect(container).toMatchSnapshot();
  });
});
