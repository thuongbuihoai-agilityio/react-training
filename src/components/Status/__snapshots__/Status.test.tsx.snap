// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component Matches snapshot 1`] = `
<div>
  <div
    class=" undefined"
    data-testid="status"
  />
</div>
`;
